tensorflow_probability/python/bijectors/real_nvp.py

```diff
@@ -171,20 +181,36 @@ def __init__(self,
       name: Python `str`, name given to ops managed by this object.
 
     Raises:
-      ValueError: If num_masked < 0.
       ValueError: If both or none of `shift_and_log_scale_fn` and `bijector_fn`
           are specified.
     """
     name = name or 'real_nvp'
 
-    self._num_masked = num_masked
-
-    self._reverse_mask = self._num_masked < 0
     # At construction time, we don't know input_depth.
     self._input_depth = None
-    if bool(shift_and_log_scale_fn) == bool(bijector_fn):
-      raise ValueError('Exactly one of `shift_and_log_scale_fn` and '
-                       '`bijector_fn` should be specified.')
+    if num_masked is not None and fraction_masked is not None:
+      raise ValueError("Exactly one of `num_masked` and "
+                       "`fraction_masked` should be specified.")
+
+    if num_masked is not None:
+      assert int(num_masked) == num_masked, (
+          "`num_masked` should be an integer. Got: {} of type {}"
+          "".format(num_masked, type(num_masked)))
+      self._num_masked = int(num_masked)
+      self._fraction_masked = None
+      self._reverse_mask = self._num_masked < 0
+    else:
+      assert np.issubdtype(type(fraction_masked), np.floating), (
+          "`fraction_masked` should be a float. Got: {} of type {}"
+          "".format(fraction_masked, type(fraction_masked)))
+      self._num_masked = None
+      self._fraction_masked = float(fraction_masked)
+      self._reverse_mask = self._fraction_masked < 0
+
+    if shift_and_log_scale_fn is not None and bijector_fn is not None:
```

Comment:
Note that these changes are not strictly needed for this PR. But I think this way of checking the `None`s is better practice than what it used to be (`bool(shift_and_log_scale_fn) == bool(bijector_fn)`).